mlrun/runtimes/nuclio/application/application.py

```diff
@@ -327,6 +327,14 @@ def deploy(
         self._ensure_reverse_proxy_configurations(self)
         self._configure_application_sidecar()
 
+        if (
```

Comment:
Validations are generally preformed in the server side. Also this validation is misplaced. You are validating the API gateway in the application runtime so there is mixup in responsibility. The validation should be in the server create api gateway endpoint. I understand the need to fail fast but if deploy was successful and application gateway creation failed - users can create the api gateway without deploying the function again so it is not a big blow.

Reply:
@alonmr in api gateway you can't not to pass creds:
![image](https://github.com/user-attachments/assets/0f09a19b-4fb6-437c-9714-0577227c2ef7)

empty-string credentials are still credentials from Nuclio point of view.